clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -848,6 +848,8 @@ public void handleResponse(AbstractResponse response) {
                         coordinatorReloaded = true;
                         lookupCoordinator(FindCoordinatorRequest.CoordinatorType.GROUP, builder.consumerGroupId());
                     }
+                } else if (error == Errors.UNKNOWN_TOPIC_OR_PARTITION) {
```

Comment:
@guozhangwang not sure I follow: Only the handlers for these two request types will treat an `UNKNOWN_TOPIC_OR_PARTITION` response as a retriable error. All the other handlers will mark the transaction in error state if this response occurs, and then the only option open to the application is to abort the transaction.

Do you mean to do additional validation that mapping of callback to request is indeed correct?